spec/mailers/user_notifications_spec.rb

```diff
@@ -1453,6 +1307,71 @@ def expects_build_with(condition)
     end
   end
 
+  describe ".notification_email" do
+    let!(:plugin) { Plugin::Instance.new }
+    let(:response_by_user) { Fabricate(:user, name: "John Doe") }
+    let(:category) { Fabricate(:category, name: "India") }
+
+    let(:topic) { Fabricate(:topic, category: category, title: "Super cool topic") }
+
+    let(:user) { Fabricate(:user) }
+    let(:post) { Fabricate(:post, topic: topic, raw: "This is My super duper cool topic") }
+    let(:response) do
+      Fabricate(
+        :basic_reply,
+        topic: post.topic,
+        user: response_by_user,
+        raw: "@#{user.username} response to post",
+      )
+    end
+    let(:notification) { Fabricate(:mentioned_notification, user: user, post: response) }
+    let!(:modify_post) do
+      post.cooked = "modified post"
+      Proc.new do
+        {
+          title: "Super cool topic",
+          post: post,
+          username: "bruce4",
+          from_alias: "John Doe via Discourse",
+          allow_reply_by_email: true,
+          use_site_subject: true,
+          add_re_to_subject: nil,
+          show_category_in_subject: true,
+          show_tags_in_subject: true,
+          show_group_in_subject: nil,
+          notification_type: "mentioned",
+          use_invite_template: nil,
+          use_topic_title_subject: false,
+          user: user,
+        }
+      end
+    end
```

Comment:
The proc passed to the modifier doesn't reflect how the modifier will be used by plugins. The `proc` needs to accept the `email_options` as a param and modify it.